sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala

```diff
@@ -3237,6 +3237,20 @@ class Dataset[T] private[sql](
     files.toSet.toArray
   }
 
+  def collectCountAndIterator(): (Long, Iterator[T]) =
```

Comment:
Would it be good to make it public?

Reply:
yea, why is it public

Reply:
I think it would be good to make public. There is no "action" to decode rows incrementally except toLocalIterator. toLocalIterator has poor performance for data sources that have many partitions or for selecting rows with limited count. So it would be good choice providing another option for reducing memory pressure of decompressing & deserializing result rows.

Reply:
We can always make it public later if there is such requirement. We should be careful to add public api.

Reply:
Add description to collectCountAndIterator, but don't make it public or experimental yet.